python/helpers/pydev/_pydevd_bundle/pydevd_save_locals.py

```diff
@@ -32,32 +22,48 @@ def save_locals(frame):
     Note: the 'save_locals' branch had a different approach wrapping the frame (much more code, but it gives ideas
     on how to save things partially, not the 'whole' locals).
     """
-    from _pydevd_bundle import pydevd_vars
-    if not isinstance(frame, pydevd_vars.frame_type):
+    if not isinstance(frame, frame_type):
         # Fix exception when changing Django variable (receiving DjangoTemplateFrame)
         return
 
+    if save_locals_impl is not None:
+        try:
+            save_locals_impl(frame)
+        except:
+            pass
+
+
+def make_save_locals_impl():
+    """
+    Factory for the 'save_locals_impl' method. This may seem like a complicated pattern but it is essential that the method is created at
+    module load time. Inner imports after module load time would cause an occasional debugger deadlock due to the importer lock and debugger
+    lock being taken in different order in  different threads.
+    """
     try:
         if '__pypy__' in sys.builtin_module_names:
             import __pypy__  # @UnresolvedImport
             save_locals = __pypy__.locals_to_fast
-            save_locals(frame)
-            return
     except:
         pass
+    else:
+        if '__pypy__' in sys.builtin_module_names:
+            def save_locals_pypy_impl(frame):
+                save_locals(frame)
```

Comment:
It's better to use another name for `locals_to_fast`, because we already have `save_locals` function in this file.